src/app/js/containers/files-explorer.js

```diff
@@ -1,32 +1,127 @@
-import React, {PropTypes, Component} from 'react'
-import {Row, Col} from 'react-bootstrap'
+import React, {Component} from 'react'
+import PropTypes from 'prop-types'
 import {connect} from 'react-redux'
-import {join} from 'path'
+import {join, dirname, basename} from 'path'
 import {includes} from 'lodash-es'
 import {toastr} from 'react-redux-toastr'
-
-import {files, router} from '../actions'
+import {withRouter} from 'react-router'
+import {pages, files} from '../actions'
 
 import Tree from './../components/files/tree'
 import ActionBar from './../components/files/action-bar'
 import Breadcrumbs from './../components/files/breadcrumbs'
 
 class FilesExplorer extends Component {
-  _onRowClick = (file, shiftKey) => {
+  componentWillMount () {
+    this.props.load()
+  }
+
+  componentDidMount () {
+    const path = this.props.match.params[0]
+    const {setRoot} = this.props
+
+    if (path) {
+      setRoot(join('/', path))
+    } else {
+      setRoot('/')
+    }
+
+    document.addEventListener('keydown', this._onKeyDown)
+  }
+
+  componentWillReceiveProps (nextProps) {
+    const locationChanged = nextProps.location !== this.props.location
+    const {setRoot} = this.props
+
+    if (locationChanged) {
+      setRoot(join('/', nextProps.match.params[0]))
+    }
+  }
+
+  componentWillUnmount () {
+    document.removeEventListener('keydown', this._onKeyDown)
+    this.props.leave()
+  }
+
+  _onKeyDown = (event) => {
+    const {deselectAll} = this.props
+
+    switch (event.which) {
+      case 27:
+        // Escape
+        event.preventDefault()
+        deselectAll()
+        break
+      case 46:
+        // Delete
+        event.preventDefault()
+        this._onRemoveDir()
+        break
+      case 113:
+        // F2
+        event.preventDefault()
+        this._onMoveDir()
+        break
+    }
+  }
+
+  selectRangeFromCurrent = (toFile) => {
+    const {
+      selected,
+      list
+    } = this.props
+
+    let fi = 0
+    let la = list.indexOf(toFile)
+
+    for (let i = 0; i < list.length; i++) {
+      if (list[i].Name !== basename(selected[0])) {
+        continue
+      }
+
+      if (la > i) {
+        fi = i + 1
+      } else {
+        fi = la
+        la = i - 1
+      }
+      break
+    }
+
+    this.selectAllBetween(fi, la)
```

Comment:
@hacdias can we convert this loop into something more JavaScript'y/Functional? Check out all the beautiful array methods https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array and try to use variable names that try to be more explicit (it is ok to use full words in JS land)